### DISCUSSION TITLE
How to detect video language without conversion?

### DISCUSSION THREAD

**Discussion by schotek** [openai/whisper#733] [Q&A] [answered]
Hey. Hey,

Whisper's good at detecting language if I don't give it one. But is it possible to only run this detection without it then starting to convert the video to that language?

**Comment by glangford** [comment, marked as answer]
The README shows how to run detection only. Scroll down to "Python usage" near the bottom, starting with

> Below is an example usage of whisper.detect_language() and whisper.decode() which provide lower-level access to the model.

```python
# load audio and pad/trim it to fit 30 seconds
audio = whisper.load_audio("audio.mp3")
audio = whisper.pad_or_trim(audio)

# make log-Mel spectrogram and move to the same device as the model
mel = whisper.log_mel_spectrogram(audio).to(model.device)

# detect the spoken language
_, probs = model.detect_language(mel)
print(f"Detected language: {max(probs, key=probs.get)}")
```
